### PR TITLE
Issue #5604: resolve Sonar violation 'Use the primitive boolean expression here'

### PR DESCRIPTION
**Pull request by romani**
Issue #5604

https://sonarcloud.io/project/issues?id=org.checkstyle%3Acheckstyle&issues=AW9t2w0PYD2QG1pPXITg&open=AW9t2w0PYD2QG1pPXITg
Should gone.

### CODE REVIEW

**Comment by rnveach** [comment]
@romani commit message is missing `#`.

**Comment by romani** [comment]
fixed.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/NPathComplexityCheck.java [hunk 1]
```diff
@@ -343,7 +343,7 @@ private void visitUnitaryOperator(DetailAST ast, int basicBranchingFactor) {
      * Leaves ternary operator (?:) and return tokens.
      */
     private void leaveUnitaryOperator() {
-        if (!afterValues.pop()) {
+        if (Boolean.FALSE.equals(afterValues.pop())) {
```
Comment (rnveach) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/NPathComplexityCheck.java]: What about `!afterValues.pop().booleanValue()`?

Comment (romani) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/NPathComplexityCheck.java]: Sonar do not like the fact that `afterValues.pop()` might be null, so your code still prone for NPE.

Comment (rnveach) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/NPathComplexityCheck.java]: Old code was prone to NPE as this is basically what it was converting the code to. If we had an issue with NPE here, regression would have found it.

Comment (romani) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/NPathComplexityCheck.java]: we do not have issue, it is the same over caution as https://checkstyle.org/config_coding.html#EqualsAvoidNull , .... if code is changed somewhere in wrong way at least this line will not be stable.

**Pull request merged**